Frameworks/UIKit/UIButton.mm

```diff
@@ -456,42 +456,83 @@ - (BOOL)isEnabled {
 
 /**
  @Status Interoperable
- @Notes The xaml element may be modified directly and we could still return stale values.
 */
-- (void)setBackgroundImage:(UIImage*)image forState:(UIControlState)state {
-    _states[state].backgroundImage = image;
+- (void)setImage:(UIImage*)image forState:(UIControlState)state {
+    _states[state].image = image;
+
+    // NOTE: check if image is nil before creating inspectableImage
+    // ConvertUIImageToWUXMImageBrush:nil creates a valid imageBrush with null comObj
+    // which isn't what we want
+    if (image) {
+        WUXMImageBrush* imageBrush = XamlUtilities::ConvertUIImageToWUXMImageBrush(image);
+        if (imageBrush) {
+            _states[state].inspectableImage = [imageBrush comObj];
+        }
+    } else {
+        // this enforces the fallback of using Image of normalState
+        // when a image for other states does not exis
+        _states[state].inspectableImage = nullptr;
+    }
+
+    // Update the Xaml elements immediately, so the image proxy reflects reality
+    XamlButtonApplyVisuals([_xamlButton comObj],
+                           _currentInspectableImage(self),
+                           _currentInspectableBorderBackgroundBrush(self));
 
     [self invalidateIntrinsicContentSize];
     [self setNeedsLayout];
 }
 
 /**
- @Status Interoperable
- @Notes The xaml element may be modified directly and we could still return stale values.
+@Status Interoperable
+@Notes The xaml element may be modified directly and we could return stale values.
 */
-- (UIImage*)backgroundImageForState:(UIControlState)state {
-    return _states[state].backgroundImage;
+-(UIImage*)imageForState:(UIControlState)state {
+    const auto& statePair = _states.find(state);
```

Comment:
maybe could use a comment.  a subtle change that avoids inserting empty values in to the map (which wastes space) when retrieving empty values